Managers/WindowMan.cpp

```diff
@@ -0,0 +1,602 @@
+
+		// Move the window to the detected leftmost display to avoid all the headaches.
+		if (m_DisplayIndexPrimaryWindowIsAt != m_DisplayArrangmentLeftMostDisplayIndex) {
+			SDL_SetWindowPosition(m_PrimaryWindow.get(), SDL_WINDOWPOS_CENTERED_DISPLAY(m_DisplayArrangmentLeftMostDisplayIndex), SDL_WINDOWPOS_CENTERED_DISPLAY(m_DisplayArrangmentLeftMostDisplayIndex));
+			m_DisplayIndexPrimaryWindowIsAt = m_DisplayArrangmentLeftMostDisplayIndex;
+		}
+
+		bool errorSettingFullscreen = false;
+
+		for (const auto &[displayIndex, displayBounds] : m_ValidDisplayIndicesAndBoundsForMultiDisplayFullscreen) {
+			int displayOffsetX = displayBounds.x;
+			int displayOffsetY = displayBounds.y;
+			int displayWidth = displayBounds.w;
+			int displayHeight = displayBounds.h;
+
+			if (displayIndex == m_DisplayIndexPrimaryWindowIsAt) {
+				m_MultiDisplayWindows.emplace_back(m_PrimaryWindow);
+				m_MultiDisplayRenderers.emplace_back(m_PrimaryRenderer);
+			} else {
+				m_MultiDisplayWindows.emplace_back(SDL_CreateWindow(nullptr, displayOffsetX, displayOffsetY, displayWidth, displayHeight, SDL_WINDOW_FULLSCREEN_DESKTOP | SDL_WINDOW_SKIP_TASKBAR), SDLWindowDeleter());
+				if (m_MultiDisplayWindows.back()) {
+					m_MultiDisplayRenderers.emplace_back(SDL_CreateRenderer(m_MultiDisplayWindows.back().get(), -1, SDL_RENDERER_ACCELERATED), SDLRendererDeleter());
+					if (!m_MultiDisplayRenderers.back()) {
+						errorSettingFullscreen = true;
+					}
+				} else {
+					errorSettingFullscreen = true;
+				}
+				if (errorSettingFullscreen) {
+					break;
+				}
+			}
+
+			int textureOffsetX = (displayOffsetX - m_DisplayArrangementLeftMostOffset);
+			int textureOffsetY = (displayOffsetY - m_DisplayArrangementTopMostOffset);
+
+			m_MultiDisplayTextureOffsets.emplace_back(SDL_Rect{
+				textureOffsetX / resMultiplier,
+				// Sometimes an odd Y offset implodes all the things, depending on the stupidity of the arrangement and what display is primary.
+				// Sometimes it needs to be in multiples of 4 for reasons unknown to man, so we're just gonna go with this and hope for the best, for now at least.
+				RoundToNearestMultiple(textureOffsetY, 2) / resMultiplier,
+				displayWidth / resMultiplier,
+				displayHeight / resMultiplier
+			});
+		}
+
+		if (errorSettingFullscreen) {
+			ClearMultiDisplayData();
+			SDL_SetWindowPosition(m_PrimaryWindow.get(), windowPrevPositionX, windowPrevPositionY);
+			return false;
+		}
+
+		CreateMultiDisplayTextures();
+		SDL_SetWindowFullscreen(m_PrimaryWindow.get(), SDL_WINDOW_FULLSCREEN_DESKTOP);
+
+		return true;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void WindowMan::DisplaySwitchIn() const {
+		g_UInputMan.DisableMouseMoving(false);
+		if (!m_MultiDisplayWindows.empty()) {
+			SDL_RaiseWindow(m_PrimaryWindow.get());
+			for (auto &window : m_MultiDisplayWindows) {
```

Comment:
I'm surprised you're using auto for something as simple as `const shared_ptr<SDL_Window> &`